lib/ssd1306.c

```diff
@@ -39,17 +39,25 @@ uint8_t ssd1306_Init(I2C_HandleTypeDef *hi2c)
     status += ssd1306_WriteCommand(hi2c, 0xFF);
     status += ssd1306_WriteCommand(hi2c, 0xA1);   // Set segment re-map 0 to 127
     status += ssd1306_WriteCommand(hi2c, 0xA6);   // Set normal display
+
     status += ssd1306_WriteCommand(hi2c, 0xA8);   // Set multiplex ratio(1 to 64)
-    status += ssd1306_WriteCommand(hi2c, 0x3F);
+    status += ssd1306_WriteCommand(hi2c, SSD1306_HEIGHT - 1);
+
     status += ssd1306_WriteCommand(hi2c, 0xA4);   // 0xa4,Output follows RAM content;0xa5,Output ignores RAM content
     status += ssd1306_WriteCommand(hi2c, 0xD3);   // Set display offset
     status += ssd1306_WriteCommand(hi2c, 0x00);   // No offset
     status += ssd1306_WriteCommand(hi2c, 0xD5);   // Set display clock divide ratio/oscillator frequency
     status += ssd1306_WriteCommand(hi2c, 0xF0);   // Set divide ratio
     status += ssd1306_WriteCommand(hi2c, 0xD9);   // Set pre-charge period
     status += ssd1306_WriteCommand(hi2c, 0x22);
+
     status += ssd1306_WriteCommand(hi2c, 0xDA);   // Set com pins hardware configuration
-    status += ssd1306_WriteCommand(hi2c, 0x12);
+#ifndef SSD1306_COM_LR_DISABLE_REMAP
+    status += ssd1306_WriteCommand(hi2c, 0x12);   // Enable COM left/right remap
```

Comment:
Bit 5 is left/right remap. Bit 4 is alternative (interlaced) COM pins configuration. What you are switching is Bit 4, not Bit 5. Otherwise, the code is correct.

Reply:
On my 128x32 display I had to change it to 0x02.